src/data/episode.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python3
+import os
+import time
+import logging
+
+import numpy as np
+
+
+class Episode(object):
+    def __init__(self, support, query):
+        self.support = support
+        self.query = query
+
+
+class SQSampler(object):
+    """A sampler for randomly sampling support/query sets.
+
+    Arguments:
+        root (str): the root of the data directory
+        support_size (int): number of songs in the support set
+        query_size (int): number of songs in the query set
+        cache (bool): caches the song names instead of hitting the FS
+    """
+    def __init__(self, root, support_size, query_size, cache=True):
+        self.root = root
+        self.support_size = support_size
+        self.query_size = query_size
+        self.cache = cache
+        if cache:
+            self.songs = {}
+
+    def sample(self, artist):
+        if self.cache and artist in self.songs:
+            songs = self.songs[artist]
+        else:
+            artist_path = os.path.join(self.root, artist)
+            songs = os.listdir(artist_path)
+            songs = [song for song in songs if not os.path.isdir(song)]
+            if self.cache:
+                self.songs[artist] = songs
+        sample = np.random.choice(
+            songs,
+            size=self.support_size+self.query_size,
+            replace=False)
+        query = sample[:self.query_size]
+        support = sample[self.query_size:]
+        return query, support
+
+
+class EpisodeSampler(object):
```

Comment:
It would be great if you could add a function that picks up the relevant parameters from a config file and returns the `EpisodeSampler` object. Right now you need to create a `LyricsLoader`, then a `Dataset` and finally the `EpisodeSampler`. If this process is automated, the user just needs to change the config to run another experiment.

Reply:
I just pushed a function to create an EpisodeSampler from a config!